pkg/validator/controllers/generic.go

```diff
@@ -0,0 +1,126 @@
+package controllers
+
+import (
+	"time"
+
+	"github.com/fairwindsops/polaris/pkg/config"
+	"github.com/sirupsen/logrus"
+	kubeAPICoreV1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	kubeAPIMetaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// GenericController is a base implementation with some free methods for inherited structs
+type GenericController struct {
+	Name        string
+	Namespace   string
+	PodSpec     kubeAPICoreV1.PodSpec
+	ObjectMeta  kubeAPIMetaV1.ObjectMeta
+	Kind        config.SupportedController
+	KindString  string
```

Comment:
I don't like the idea of tracking both `Kind` and `KindString` - it's not immediately apparent why they'd diverge (though I can see from the code that they do)

Is there anything wrong with always setting `Kind` to the controller Kind, and getting rid of the `SupportedController` type?

Reply:
I got rid of the Kind on the GenericController, but left the SupportedController type because it still seemed useful for a few things, especially it seems like we use it on the validation webhook side.